### PR TITLE
expose public method to generate a s2a-enabled tls configuration

### PR DESCRIPTION
**Pull request by xmenxk**
bazel build ...
bazel test ...

### CODE REVIEW

File: s2a.go [hunk 1]
```diff
@@ -276,6 +278,52 @@ func (c *s2aTransportCreds) OverrideServerName(serverNameOverride string) error
...
+// TLSClientConfigOptions specifies parameters for creating client TLS config.
+type TLSClientConfigOptions struct{}
+
+// TLSClientConfigFactory defines the interface for a client TLS config factory.
+type TLSClientConfigFactory interface {
+	Build(ctx context.Context, opts *TLSClientConfigOptions) (*tls.Config, error)
+}
+
+// NewTLSClientConfigFactory returns an instance of s2aTLSClientConfigFactory.
+func NewTLSClientConfigFactory(opts *ClientOptions) (TLSClientConfigFactory, error) {
```
Comment (matthewstevenson88) [inline s2a.go:290]: Can we add tests for this API, similar to what you did in s2av2_e2e_test.go?

**Review by matthewstevenson88 — COMMENTED** [review]
Thanks for the PR Kui! Looks good, just a request for some more tests. :)

**Comment by matthewstevenson88** [comment]
Also it looks like one of the presubmits failed - can you fix whatever the test that this PR breaks?

File: s2a.go (hunk 1 above)
Comment (xmenxk) [reply s2a.go:290]: done, thanks!

**Comment by xmenxk** [comment]
> Also it looks like one of the presubmits failed - can you fix whatever the test that this PR breaks?

seems like a transient issue, all passing now, PTAL :)

**Review by matthewstevenson88 — APPROVED** [review]
Thanks for the PR Kui!

**Comment by matthewstevenson88** [comment]
Please "squash and merge" rather than just merging.

**Pull request merged**